pipelines/resource_pipeline.go

```diff
@@ -199,12 +212,22 @@ func (a PipelinesAPI) Create(s PipelineSpec, timeout time.Duration) (string, err
 	return id, nil
 }
 
+func adjustForceSendFields(s *PipelineSpec) {
+	for i := range s.Clusters {
+		cluster := &s.Clusters[i]
+		if cluster.SparkConf["spark.databricks.cluster.profile"] == "singleNode" {
+			cluster.ForceSendFields = append(cluster.ForceSendFields, "NumWorkers")
```

Comment:
Can't we always send this instead? My understanding is that if it's not explicitly set it won't be sent

Reply:
TF SDK doesn't have a method to know if a field is set or not. In this case, not set means "use the default autostaling, with min 2- max 5 workers". While NumWorkers 0 means use 0 workers.

We are instead using this spark config which tells Spark that there are no workers. This config only makes sense if NumWorkers is zero, so we use it to determine that we need to send the field.

I will add a comment.

Reply:
What would happens if we mark NumWorkers as ForceSendFields always without the condition?
In particular if it's not set in TF but marked as ForceSendField, which value will be sent?

Reply:
Then we would send 0, which is the default GoLang value for Int.